common/utils/src/main/resources/error/error-classes.json

```diff
@@ -2304,6 +2304,12 @@
     },
     "sqlState" : "22023"
   },
+  "INVALID_PARTITION_COLUMN_DATA_TYPE" : {
+    "message" : [
+      "Cannot use <type> for partition column."
+    ],
+    "sqlState" : "42601"
```

Comment:
Actually the 0A000 was more correct.
0A000 means a feature is not supported. It implies that in some future it may be and the user hit a limit rather than just writing something inherently wrong.

Reply:
thanks for the response! will fix it in a separate PR since this one is already closed

Reply:
Also let's stay consistent with error messages. E.g. in [string functions PR ](https://github.com/apache/spark/pull/45216) @uros-db is also using `0A000` for as `sqlState` so let us align there.